pkg/cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -121,8 +121,13 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 	}
 
 	defer client.Close()
-
-	return deleteVolumeAndDomain(machine, client)
+	exists, err := libvirtutils.DomainExists(machine.Name, client)
+	if exists {
+		return deleteVolumeAndDomain(machine, client)
+	} else {
```

Comment:
don't need the `else` here right?

Reply:
log message is being printed only in else case, thats the case when we are skipping deletion

Reply:
yeh, still think don't need the `else` right? It'll only log when `exist` is false 
how about
```
	exists, err := libvirtutils.DomainExists(machine.Name, client)
	if err != nil {
		return err
	}
	if exists {
		return deleteVolumeAndDomain(machine, client)
	}
	glog.Infof("Machine does not exist. Skipping deletion...")
	return nil
```

Reply:
i am so dumb. you are right :D